src/pickup.cpp

```diff
@@ -567,6 +567,15 @@ void Pickup::pick_up( const tripoint &p, int min, from_where get_items_from )
         int pickupH = maxitems + pickupBorderRows;
         int pickupW = 44;
 
+        //find max length of item name and resize pickup window width
+        for( size_t cur_it = 0; cur_it < stacked_here.size(); cur_it++ ) {
+            const item &this_item = *stacked_here[cur_it].front();
```

Comment:
`index` in your code is only used to access the elements of `stacked_here`, so it can be simplified to a range-based for loop. You only need to change
```
        for( size_t cur_it = 0; cur_it < stacked_here.size(); cur_it++ ) {
            const item &this_item = *stacked_here[cur_it].front();
```
to
```
        for( const std::list<item_stack::iterator> &cur_list : stacked_here ) {
            const item &this_item = *cur_list.front();
```